example/hr-data-generator/src/main/java/uk/gov/gchq/palisade/example/hrdatagenerator/types/Address.java

```diff
@@ -17,18 +17,30 @@
 package uk.gov.gchq.palisade.example.hrdatagenerator.types;
 
 import com.github.javafaker.Faker;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
 
 import java.util.Random;
 
 public class Address {
-
     private String streetAddressNumber;
     private String streetName;
     private String city;
     private String state;
     private String zipCode;
 
+    public Address() {
+    }
+
+    public Address(final Address address) {
+        streetAddressNumber = address.streetAddressNumber;
+        streetName = address.streetName;
+        city = address.city;
+        state = address.state;
+        zipCode = address.zipCode;
+    }
```

Comment:
Strings are immutable in Java so when you do 
String a = "temp"
String b = "xyz"
a = b
This actually means that a is pointing to the reference b rather than a's contents being modified

For the above you should really do the following:
```
streetAddressNumber = new String(address.streetAddressNumber)
```